pyunittests-pyroot-pyz-ttree-setbranchaddress segfaults with Python 3.13

### Check duplicate issues.

- [X] Checked for duplicates

### Description

The pyunittests-pyroot-pyz-ttree-setbranchaddress test segfaults with Python 3.13:
```
  41/1342 Test   #31: pyunittests-pyroot-pyz-ttree-setbranchaddress .............................***Failed   57.97 sec
test_array_branch (ttree_setbranchaddress.TTreeSetBranchAddress.test_array_branch) ... ok
test_basic_type_branch (ttree_setbranchaddress.TTreeSetBranchAddress.test_basic_type_branch) ... ok
test_class_with_array_member (ttree_setbranchaddress.TTreeSetBranchAddress.test_class_with_array_member) ... ok
test_fallback_case (ttree_setbranchaddress.TTreeSetBranchAddress.test_fallback_case) ...  *** Break *** segmentation violation



===========================================================
There was a crash.
This is the entire stack trace of all threads:
===========================================================
#0  0x00007f3f13f12e87 in wait4 () from /lib64/libc.so.6
#1  0x00007f3f13e664fb in do_system () from /lib64/libc.so.6
#2  0x00007f3f13322cf0 in TUnixSystem::Exec (this=0x5619b6bb82b0, shellcmd=<optimized out>) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/core/unix/src/TUnixSystem.cxx:2151
#3  TUnixSystem::StackTrace (this=0x5619b6bb82b0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/core/unix/src/TUnixSystem.cxx:2442
#4  0x00007f3f13511c5f in (anonymous namespace)::do_trace (sig=1) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/cppyy-backend/clingwrapper/src/clingwrapper.cxx:189
#5  (anonymous namespace)::TExceptionHandlerImp::HandleException (this=<optimized out>, sig=1) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/cppyy-backend/clingwrapper/src/clingwrapper.cxx:202
#6  0x00007f3f133203b9 in TUnixSystem::DispatchSignals (this=0x5619b6bb82b0, sig=kSigSegmentationViolation) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/core/unix/src/TUnixSystem.cxx:3657
#7  <signal handler called>
#8  CPyCppyy::VoidArrayConverter::SetArg (this=0x5619bcb4fd10, pyobject=<optimized out>, para=..., ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/Converters.cxx:1513
#9  0x00007f3f00e7a068 in CPyCppyy::CPPMethod::ConvertAndSetArgs (this=this
entry=0x5619bc8a1b80, args=0x7f3f00328a98, nargsf=<optimized out>, ctxt=ctxt
entry=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:934
#10 0x00007f3f00e7a23c in CPyCppyy::CPPMethod::Call (this=0x5619bc8a1b80, self=
0x7fffe41c52d8: 0x7f3eff3870d0, args=<optimized out>, nargsf=<optimized out>, kwds=<optimized out>, ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:991
#11 0x00007f3f00e7f540 in CPyCppyy::(anonymous namespace)::mp_vectorcall (pymeth=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPOverload.cxx:699
#12 0x00007f3f00eb7cde in CPyCppyy_tp_call (cb=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPyCppyy.h:321
#13 CPyCppyy::SelectAndForward (pytmpl=pytmpl
entry=0x7f3e9b93ef70, pymeth=0x7f3f1384ce80, args=args
entry=0x7f3f00328a98, nargsf=nargsf
entry=3, kwds=kwds
entry=0x0, implicitOkay=implicitOkay
entry=true, use_targs=false, sighash=9465071328407200440, errors=...) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/TemplateProxy.cxx:481
#14 0x00007f3f00eba9fd in CPyCppyy::tpp_vectorcall (pytmpl=0x7f3e9b93ef70, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /usr/include/c++/14/bits/shared_ptr_base.h:1666
#15 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#16 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#17 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#18 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#19 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#20 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#21 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#22 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#23 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#24 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#25 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#26 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#27 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#28 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#29 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#30 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#31 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#32 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#33 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#34 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#35 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#36 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#37 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#38 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#39 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#40 0x00007f3f141a5d0b in slot_tp_init () from /lib64/libpython3.13.so.1.0
#41 0x00007f3f1415e9c5 in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#42 0x00007f3f1417d188 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#43 0x00007f3f1423f8eb in PyEval_EvalCode () from /lib64/libpython3.13.so.1.0
#44 0x00007f3f1425afe4 in builtin_exec () from /lib64/libpython3.13.so.1.0
#45 0x00007f3f1418e3e7 in cfunction_vectorcall_FASTCALL_KEYWORDS () from /lib64/libpython3.13.so.1.0
#46 0x00007f3f14160f57 in PyObject_Vectorcall () from /lib64/libpython3.13.so.1.0
#47 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#48 0x00007f3f142797ed in pymain_run_module () from /lib64/libpython3.13.so.1.0
#49 0x00007f3f1427500c in Py_RunMain () from /lib64/libpython3.13.so.1.0
#50 0x00007f3f1422bcbc in Py_BytesMain () from /lib64/libpython3.13.so.1.0
#51 0x00007f3f13e3c1c8 in __libc_start_call_main () from /lib64/libc.so.6
#52 0x00007f3f13e3c28b in __libc_start_main_impl () from /lib64/libc.so.6
#53 0x00005619b4d2b095 in _start ()
===========================================================


The lines below might hint at the cause of the crash. If you see question
marks as part of the stack trace, try to recompile with debugging information
enabled and export CLING_DEBUG=1 environment variable before running.
You may get help by asking at the ROOT forum https://root.cern/forum
preferably using the command (.forum bug) in the ROOT prompt.
Only if you are really convinced it is a bug in ROOT then please submit a
report at https://root.cern/bugs or (preferably) using the command (.gh bug) in
the ROOT prompt. Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#8  CPyCppyy::VoidArrayConverter::SetArg (this=0x5619bcb4fd10, pyobject=<optimized out>, para=..., ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/Converters.cxx:1513
#9  0x00007f3f00e7a068 in CPyCppyy::CPPMethod::ConvertAndSetArgs (this=this
entry=0x5619bc8a1b80, args=0x7f3f00328a98, nargsf=<optimized out>, ctxt=ctxt
entry=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:934
#10 0x00007f3f00e7a23c in CPyCppyy::CPPMethod::Call (this=0x5619bc8a1b80, self=
0x7fffe41c52d8: 0x7f3eff3870d0, args=<optimized out>, nargsf=<optimized out>, kwds=<optimized out>, ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:991
#11 0x00007f3f00e7f540 in CPyCppyy::(anonymous namespace)::mp_vectorcall (pymeth=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPOverload.cxx:699
#12 0x00007f3f00eb7cde in CPyCppyy_tp_call (cb=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPyCppyy.h:321
#13 CPyCppyy::SelectAndForward (pytmpl=pytmpl
entry=0x7f3e9b93ef70, pymeth=0x7f3f1384ce80, args=args
entry=0x7f3f00328a98, nargsf=nargsf
entry=3, kwds=kwds
entry=0x0, implicitOkay=implicitOkay
entry=true, use_targs=false, sighash=9465071328407200440, errors=...) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/TemplateProxy.cxx:481
#14 0x00007f3f00eba9fd in CPyCppyy::tpp_vectorcall (pytmpl=0x7f3e9b93ef70, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /usr/include/c++/14/bits/shared_ptr_base.h:1666
#15 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#16 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#17 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#18 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#19 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#20 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#21 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#22 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#23 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#24 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#25 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#26 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#27 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#28 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#29 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#30 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#31 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#32 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#33 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#34 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#35 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#36 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#37 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#38 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#39 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#40 0x00007f3f141a5d0b in slot_tp_init () from /lib64/libpython3.13.so.1.0
#41 0x00007f3f1415e9c5 in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#42 0x00007f3f1417d188 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#43 0x00007f3f1423f8eb in PyEval_EvalCode () from /lib64/libpython3.13.so.1.0
#44 0x00007f3f1425afe4 in builtin_exec () from /lib64/libpython3.13.so.1.0
#45 0x00007f3f1418e3e7 in cfunction_vectorcall_FASTCALL_KEYWORDS () from /lib64/libpython3.13.so.1.0
#46 0x00007f3f14160f57 in PyObject_Vectorcall () from /lib64/libpython3.13.so.1.0
#47 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#48 0x00007f3f142797ed in pymain_run_module () from /lib64/libpython3.13.so.1.0
#49 0x00007f3f1427500c in Py_RunMain () from /lib64/libpython3.13.so.1.0
#50 0x00007f3f1422bcbc in Py_BytesMain () from /lib64/libpython3.13.so.1.0
#51 0x00007f3f13e3c1c8 in __libc_start_call_main () from /lib64/libc.so.6
#52 0x00007f3f13e3c28b in __libc_start_main_impl () from /lib64/libc.so.6
#53 0x00005619b4d2b095 in _start ()
===========================================================


 *** Break *** segmentation violation



===========================================================
There was a crash.
This is the entire stack trace of all threads:
===========================================================
#0  0x00007f3f13f12e87 in wait4 () from /lib64/libc.so.6
#1  0x00007f3f13e664fb in do_system () from /lib64/libc.so.6
#2  0x00007f3f13322cf0 in TUnixSystem::Exec (this=0x5619b6bb82b0, shellcmd=<optimized out>) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/core/unix/src/TUnixSystem.cxx:2151
#3  TUnixSystem::StackTrace (this=0x5619b6bb82b0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/core/unix/src/TUnixSystem.cxx:2442
#4  0x00007f3f13511b7d in (anonymous namespace)::do_trace (sig=1) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/cppyy-backend/clingwrapper/src/clingwrapper.cxx:189
#5  (anonymous namespace)::TExceptionHandlerImp::HandleException (this=<optimized out>, sig=1) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/cppyy-backend/clingwrapper/src/clingwrapper.cxx:208
#6  0x00007f3f133203b9 in TUnixSystem::DispatchSignals (this=0x5619b6bb82b0, sig=kSigSegmentationViolation) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/core/unix/src/TUnixSystem.cxx:3657
#7  <signal handler called>
#8  CPyCppyy::VoidArrayConverter::SetArg (this=0x5619bcb4fd10, pyobject=<optimized out>, para=..., ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/Converters.cxx:1513
#9  0x00007f3f00e7a068 in CPyCppyy::CPPMethod::ConvertAndSetArgs (this=this
entry=0x5619bc8a1b80, args=0x7f3f00328a98, nargsf=<optimized out>, ctxt=ctxt
entry=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:934
#10 0x00007f3f00e7a23c in CPyCppyy::CPPMethod::Call (this=0x5619bc8a1b80, self=
0x7fffe41c52d8: 0x7f3eff3870d0, args=<optimized out>, nargsf=<optimized out>, kwds=<optimized out>, ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:991
#11 0x00007f3f00e7f540 in CPyCppyy::(anonymous namespace)::mp_vectorcall (pymeth=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPOverload.cxx:699
#12 0x00007f3f00eb7cde in CPyCppyy_tp_call (cb=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPyCppyy.h:321
#13 CPyCppyy::SelectAndForward (pytmpl=pytmpl
entry=0x7f3e9b93ef70, pymeth=0x7f3f1384ce80, args=args
entry=0x7f3f00328a98, nargsf=nargsf
entry=3, kwds=kwds
entry=0x0, implicitOkay=implicitOkay
entry=true, use_targs=false, sighash=9465071328407200440, errors=...) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/TemplateProxy.cxx:481
#14 0x00007f3f00eba9fd in CPyCppyy::tpp_vectorcall (pytmpl=0x7f3e9b93ef70, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /usr/include/c++/14/bits/shared_ptr_base.h:1666
#15 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#16 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#17 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#18 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#19 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#20 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#21 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#22 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#23 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#24 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#25 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#26 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#27 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#28 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#29 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#30 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#31 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#32 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#33 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#34 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#35 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#36 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#37 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#38 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#39 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#40 0x00007f3f141a5d0b in slot_tp_init () from /lib64/libpython3.13.so.1.0
#41 0x00007f3f1415e9c5 in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#42 0x00007f3f1417d188 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#43 0x00007f3f1423f8eb in PyEval_EvalCode () from /lib64/libpython3.13.so.1.0
#44 0x00007f3f1425afe4 in builtin_exec () from /lib64/libpython3.13.so.1.0
#45 0x00007f3f1418e3e7 in cfunction_vectorcall_FASTCALL_KEYWORDS () from /lib64/libpython3.13.so.1.0
#46 0x00007f3f14160f57 in PyObject_Vectorcall () from /lib64/libpython3.13.so.1.0
#47 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#48 0x00007f3f142797ed in pymain_run_module () from /lib64/libpython3.13.so.1.0
#49 0x00007f3f1427500c in Py_RunMain () from /lib64/libpython3.13.so.1.0
#50 0x00007f3f1422bcbc in Py_BytesMain () from /lib64/libpython3.13.so.1.0
#51 0x00007f3f13e3c1c8 in __libc_start_call_main () from /lib64/libc.so.6
#52 0x00007f3f13e3c28b in __libc_start_main_impl () from /lib64/libc.so.6
#53 0x00005619b4d2b095 in _start ()
===========================================================


The lines below might hint at the cause of the crash. If you see question
marks as part of the stack trace, try to recompile with debugging information
enabled and export CLING_DEBUG=1 environment variable before running.
You may get help by asking at the ROOT forum https://root.cern/forum
preferably using the command (.forum bug) in the ROOT prompt.
Only if you are really convinced it is a bug in ROOT then please submit a
report at https://root.cern/bugs or (preferably) using the command (.gh bug) in
the ROOT prompt. Please post the ENTIRE stack trace
from above as an attachment in addition to anything else
that might help us fixing this issue.
===========================================================
#8  CPyCppyy::VoidArrayConverter::SetArg (this=0x5619bcb4fd10, pyobject=<optimized out>, para=..., ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/Converters.cxx:1513
#9  0x00007f3f00e7a068 in CPyCppyy::CPPMethod::ConvertAndSetArgs (this=this
entry=0x5619bc8a1b80, args=0x7f3f00328a98, nargsf=<optimized out>, ctxt=ctxt
entry=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:934
#10 0x00007f3f00e7a23c in CPyCppyy::CPPMethod::Call (this=0x5619bc8a1b80, self=
0x7fffe41c52d8: 0x7f3eff3870d0, args=<optimized out>, nargsf=<optimized out>, kwds=<optimized out>, ctxt=0x7fffe41c5300) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPMethod.cxx:991
#11 0x00007f3f00e7f540 in CPyCppyy::(anonymous namespace)::mp_vectorcall (pymeth=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPPOverload.cxx:699
#12 0x00007f3f00eb7cde in CPyCppyy_tp_call (cb=0x7f3e9b952140, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/CPyCppyy.h:321
#13 CPyCppyy::SelectAndForward (pytmpl=pytmpl
entry=0x7f3e9b93ef70, pymeth=0x7f3f1384ce80, args=args
entry=0x7f3f00328a98, nargsf=nargsf
entry=3, kwds=kwds
entry=0x0, implicitOkay=implicitOkay
entry=true, use_targs=false, sighash=9465071328407200440, errors=...) at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/bindings/pyroot/cppyy/CPyCppyy/src/TemplateProxy.cxx:481
#14 0x00007f3f00eba9fd in CPyCppyy::tpp_vectorcall (pytmpl=0x7f3e9b93ef70, args=0x7f3f00328a98, nargsf=3, kwds=0x0) at /usr/include/c++/14/bits/shared_ptr_base.h:1666
#15 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#16 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#17 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#18 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#19 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#20 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#21 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#22 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#23 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#24 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#25 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#26 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#27 0x00007f3f14173ef1 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#28 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#29 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#30 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#31 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#32 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#33 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#34 0x00007f3f141ca25d in method_vectorcall () from /lib64/libpython3.13.so.1.0
#35 0x00007f3f14178546 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#36 0x00007f3f1423d53d in _PyObject_VectorcallDictTstate () from /lib64/libpython3.13.so.1.0
#37 0x00007f3f14291f67 in slot_tp_call () from /lib64/libpython3.13.so.1.0
#38 0x00007f3f1415ea6b in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#39 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#40 0x00007f3f141a5d0b in slot_tp_init () from /lib64/libpython3.13.so.1.0
#41 0x00007f3f1415e9c5 in _PyObject_MakeTpCall () from /lib64/libpython3.13.so.1.0
#42 0x00007f3f1417d188 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#43 0x00007f3f1423f8eb in PyEval_EvalCode () from /lib64/libpython3.13.so.1.0
#44 0x00007f3f1425afe4 in builtin_exec () from /lib64/libpython3.13.so.1.0
#45 0x00007f3f1418e3e7 in cfunction_vectorcall_FASTCALL_KEYWORDS () from /lib64/libpython3.13.so.1.0
#46 0x00007f3f14160f57 in PyObject_Vectorcall () from /lib64/libpython3.13.so.1.0
#47 0x00007f3f14173894 in _PyEval_EvalFrameDefault () from /lib64/libpython3.13.so.1.0
#48 0x00007f3f142797ed in pymain_run_module () from /lib64/libpython3.13.so.1.0
#49 0x00007f3f1427500c in Py_RunMain () from /lib64/libpython3.13.so.1.0
#50 0x00007f3f1422bcbc in Py_BytesMain () from /lib64/libpython3.13.so.1.0
#51 0x00007f3f13e3c1c8 in __libc_start_call_main () from /lib64/libc.so.6
#52 0x00007f3f13e3c28b in __libc_start_main_impl () from /lib64/libc.so.6
#53 0x00005619b4d2b095 in _start ()
===========================================================


CMake Error at /builddir/build/BUILD/root-6.32.00-build/root-6.32.00/cmake/modules/RootTestDriver.cmake:232 (message):
  error code: 129
```


### Reproducer

Compile root with Python 3.13. Run tests.

### ROOT version

6.32.00

### Installation method

RPM package build from source

### Operating system

Fedora 41 (rawhide)

### Additional context

_No response_

--- a/bindings/pyroot/pythonizations/python/ROOT/_pythonization/_ttree.py
+++ b/bindings/pyroot/pythonizations/python/ROOT/_pythonization/_ttree.py
@@ -130,7 +130,7 @@ ds.SetBranchAddress('structb', ms)
 */
 '''
 
-from libROOTPythonizations import GetBranchAttr, SetBranchAddressPyz, BranchPyz
+from libROOTPythonizations import GetBranchAttr, BranchPyz
 from . import pythonization
 
 # TTree iterator
@@ -145,15 +145,66 @@ def _TTree__iter__(self):
     if bytes_read == -1:
         raise RuntimeError("TTree I/O error")
 
-def _SetBranchAddress(self, *args):
-    # Modify the behaviour if args is (const char*, void*)
-    res = SetBranchAddressPyz(self, *args)
+def _pythonize_branch_addr(branch, addr_orig):
+    """Helper for the SetBranchAddress pythonization, extracting the relevant
+    address from a Python object if possible.
+    """
+    import cppyy
+    import ctypes
 
-    if res is None:
-        # Fall back to the original implementation for the rest of overloads
-        res = self._OriginalSetBranchAddress(*args)
+    # Pythonization for cppyy proxies (of type CPPInstance)
+    if isinstance(addr_orig, cppyy._backend.CPPInstance):
 
-    return res
+        is_leaf_list = branch.IsA() is cppyy.gbl.TBranch.Class()
+
+        if is_leaf_list:
+            # If the branch is a leaf list, SetBranchAddress expects the
+            # address of the object that has the corresponding data members.
+            return ctypes.c_void_p(cppyy.addressof(instance=addr_orig, byref=False))
+
+        # Otherwise, SetBranchAddress is expecting a pointer to the address of
+        # the object, and the pointer needs to stay alive. Therefore, we create
+        # a container for the pointer and cache it in the original cppyy proxy.
+        addr_view = cppyy.gbl.array["std::intptr_t", 1]([cppyy.addressof(instance=addr_orig, byref=False)])
+
+        if not hasattr(addr_orig, "_set_branch_cached_pointers"):
+            addr_orig._set_branch_cached_pointers = []
+        addr_orig._set_branch_cached_pointers.append(addr_view)
+
+        # Finally, we have to return the address of the container
+        return ctypes.c_void_p(cppyy.addressof(instance=addr_view, byref=False))
+
+    # For NumPy arrays
+    if hasattr(addr_orig, "__array_interface__"):
+        return ctypes.c_void_p(addr_orig.__array_interface__["data"][0])
+
+    # For the builtin array library
+    if hasattr(addr_orig, "buffer_info"):
+        return ctypes.c_void_p(addr_orig.buffer_info()[0])
+
+    # We don't know how to pythonize the address parameter. return the
+    # original value one.
+    return addr_orig
+
+def _SetBranchAddress(self, bname, addr, *args, **kwargs):
+    """
+    Pythonization for TTree::SetBranchAddress.
+
+    Modify the behaviour of SetBranchAddress so that proxy references can be passed
+    as arguments from the Python side, more precisely in cases where the C++
+    implementation of the method expects the address of a pointer.
+
+    For example:
+    ```
+    v = ROOT.std.vector('int')()
+    t.SetBranchAddress("my_vector_branch", v)
+    ```
+    """
+
+    branch = self.GetBranch(bname)
+    addr = _pythonize_branch_addr(branch, addr)
+
+    return self._OriginalSetBranchAddress(bname, addr, *args, **kwargs)
 
 def _Branch(self, *args):
     # Modify the behaviour if args is one of:

--- a/bindings/pyroot/pythonizations/src/PyROOTModule.cxx
+++ b/bindings/pyroot/pythonizations/src/PyROOTModule.cxx
@@ -48,8 +48,6 @@ static PyMethodDef gPyROOTMethods[] = {
     (char *)"Cast the void* returned by TClass::DynamicCast to the right type"},
    {(char *)"AddTObjectEqNePyz", (PyCFunction)PyROOT::AddTObjectEqNePyz, METH_VARARGS,
     (char *)"Add equality and inequality comparison operators to TObject"},
-   {(char *)"SetBranchAddressPyz", (PyCFunction)PyROOT::SetBranchAddressPyz, METH_VARARGS,
-    (char *)"Fully enable the use of TTree::SetBranchAddress from Python"},
    {(char *)"BranchPyz", (PyCFunction)PyROOT::BranchPyz, METH_VARARGS,
     (char *)"Fully enable the use of TTree::Branch from Python"},
    {(char *)"AddPrettyPrintingPyz", (PyCFunction)PyROOT::AddPrettyPrintingPyz, METH_VARARGS,

--- a/bindings/pyroot/pythonizations/src/PyROOTPythonize.h
+++ b/bindings/pyroot/pythonizations/src/PyROOTPythonize.h
@@ -22,7 +22,6 @@ PyObject *AddPrettyPrintingPyz(PyObject *self, PyObject *args);
 
 PyObject *GetBranchAttr(PyObject *self, PyObject *args);
 PyObject *BranchPyz(PyObject *self, PyObject *args);
-PyObject *SetBranchAddressPyz(PyObject *self, PyObject *args);
 
 PyObject *AddTClassDynamicCastPyz(PyObject *self, PyObject *args);
 

--- a/bindings/pyroot/pythonizations/src/TTreePyz.cxx
+++ b/bindings/pyroot/pythonizations/src/TTreePyz.cxx
@@ -202,69 +202,6 @@ PyObject *PyROOT::GetBranchAttr(PyObject * /*self*/, PyObject *args)
 }
 
 ////////////////////////////////////////////////////////////////////////////
-/// \brief Add pythonization for TTree::SetBranchAddress.
-/// \param[in] self Always null, since this is a module function.
-/// \param[in] args Pointer to a Python tuple object containing the arguments
-/// received from Python.
-///
-/// Modify the behaviour of SetBranchAddress so that proxy references can be passed
-/// as arguments from the Python side, more precisely in cases where the C++
-/// implementation of the method expects the address of a pointer.
-///
-/// For example:
-/// ~~~{.py}
-/// v = ROOT.std.vector('int')()
-/// t.SetBranchAddress("my_vector_branch", v)
-/// ~~~
-PyObject *PyROOT::SetBranchAddressPyz(PyObject * /* self */, PyObject *args)
-{
-   PyObject *treeObj = nullptr, *name = nullptr, *address = nullptr;
-
-   int argc = PyTuple_GET_SIZE(args);
-
-   // Look for the (const char*, void*) overload
-   auto argParseStr = "OUO:SetBranchAddress";
-   if (argc == 3 && PyArg_ParseTuple(args, argParseStr, &treeObj, &name, &address)) {
-
-      auto tree = (TTree *)GetTClass(treeObj)->DynamicCast(TTree::Class(), CPyCppyy::Instance_AsVoidPtr(treeObj));
-
-      if (!tree) {
-         PyErr_SetString(PyExc_TypeError,
-                         "TTree::SetBranchAddress must be called with a TTree instance as first argument");
-         return nullptr;
-      }
-
-      auto branchName = PyUnicode_AsUTF8(name);
-      auto branch = tree->GetBranch(branchName);
-      if (!branch) {
-         PyErr_SetString(PyExc_TypeError, "TTree::SetBranchAddress must be called with a valid branch name");
-         return nullptr;
-      }
-
-      bool isLeafList = branch->IsA() == TBranch::Class();
-
-      void *buf = 0;
-      if (CPyCppyy::Instance_Check(address)) {
-         ((CPPInstance *)address)->GetDatamemberCache(); // force creation of cache
-
-         if (((CPPInstance *)address)->fFlags & CPPInstance::kIsReference || isLeafList)
-            buf = CPyCppyy::Instance_AsVoidPtr(address);
-         else
-            buf = (void *)&(((CPPInstance *)address)->GetObjectRaw());
-      } else
-         Utility::GetBuffer(address, '*', 1, buf, false);
-
-      if (buf != nullptr) {
-         auto res = tree->SetBranchAddress(PyUnicode_AsUTF8(name), buf);
-         return PyInt_FromLong(res);
-      }
-   }
-
-   // Not the overload we wanted to pythonize, return None
-   Py_RETURN_NONE;
-}
-
-////////////////////////////////////////////////////////////////////////////
 /// Try to match the arguments of TTree::Branch to the following overload:
 /// - ( const char*, void*, const char*, Int_t = 32000 )
 /// If the match succeeds, invoke Branch on the C++ tree with the right

